rules_csharp_gapic/csharp_compiler_repo.bzl

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _csharp_compiler_impl(ctx):
+    ctx.download_and_extract(
+        url="https://download.visualstudio.microsoft.com/download/pr/8db2b522-7fa2-4903-97ec-d6d04d297a01/f467006b9098c2de256e40d2e2f36fea/dotnet-sdk-3.1.301-linux-x64.tar.gz",
```

Comment:
We would need to support at least mac platform as well.  The proper way of doing it would be toolchains, but a simple way would be just checking the os type on the repository rule and downloading a proper binary based on that. Please check PHP implementation for an example of how to do it: https://github.com/googleapis/gapic-generator/blob/master/rules_gapic/php/php_gapic_repositories.bzl#L26